pkg/cloudprovider/vsphereparavirtual/controllers/routablepod/node/node_controller.go

```diff
@@ -199,59 +200,13 @@ func (c *Controller) syncNode(key string) error {
 	node, err := c.nodesLister.Get(name)
 	switch {
 	case apierrors.IsNotFound(err):
-		// node absence in store means watcher caught the deletion, ensure the request in ippool is deleted info is cleaned
-		err = c.processNodeDelete(name)
+		// node absence in store means watcher caught the deletion, ensure Pod CIDR of this Node is released
+		err = c.nsxIPManager.ReleasePodCIDR(node)
 	case err != nil:
 		utilruntime.HandleError(fmt.Errorf("unable to retrieve node %v from store: %v", key, err))
 	default:
-		err = c.processNodeCreateOrUpdate(node)
+		err = c.nsxIPManager.ClaimPodCIDR(node)
```

Comment:
nit: maybe add a comment here to explain that we are going to claim pod cidr when node creation